### PR TITLE
Search: fix missing dashboards due to empty uid, log errors

### PR DESCRIPTION
**Pull request by FZambia**
**What this PR does / why we need it**:

This PR improves dashboard data reading. Solves 2 problems:

* some dashboards do not have `uid` in `data` so they are missing in query result (due to permission filter check)
* some dashboards have invalid JSON in `data`. This requires more investigation - as may be an import to SQLite problem, but logging `iter.Error` is useful anyway.

Also:
* log main org index build time on start.

### CODE REVIEW

**Comment by ryantxu** [comment]
We need to land in a place where everything has a UID, and we never expose the internal ID.

Can we update this so that we get the UID from SQL and put it into the dashboard rather than removing it?

**Comment by ryantxu** [comment]
We need to land in a place where everything has a UID, and we never expose the internal ID.

Can we update this so that we get the UID from SQL and put it into the dashboard rather than removing it?

**Comment by ArturWierzbicki** [comment]
> Can we update this so that we get the UID from SQL and put it into the dashboard rather than removing it?

I think that's what is already happening in this PR: https://github.com/grafana/grafana/blob/a431427fc1fac4e94f6b4cb8ef448b4f6d8f1af5/pkg/services/searchV2/index.go#L325

Or are you asking about fixing the source data?

File: pkg/services/searchV2/index.go [hunk 1]
```diff
@@ -308,6 +315,11 @@ func (l sqlDashboardLoader) LoadDashboards(ctx context.Context, orgID int64, das
 		}
 
 		for _, row := range rows {
+			info, err := extract.ReadDashboard(bytes.NewReader(row.Data), lookup)
+			if err != nil {
+				l.logger.Debug("Error reading dashboard data", "error", err, "dashboardId", row.Id, "dashboardSlug", row.Slug)
```
Comment (ArturWierzbicki) [inline pkg/services/searchV2/index.go:320]: Should we log it on `info` instead, at least for a while? I think we won't be able to see debug logs on HG

Comment (FZambia) [reply pkg/services/searchV2/index.go:320]: Yep 👍

Comment (FZambia) [reply pkg/services/searchV2/index.go]: Logging on Warn now

**Review by ArturWierzbicki — APPROVED** [review]
🚀

**Review by ryantxu — APPROVED** [review]
I see -- we are removing UID from extract.DashboardInfo, but making sure it is alwasy there for the internal representation. 👍

**Pull request merged**